.github/workflows/maven.yml

```diff
@@ -32,3 +32,22 @@ jobs:
           restore-keys: ${{ runner.os }}-m2
       - name: Build
         run: mvn ${{ env.maven_commands }}
+  deploy:
```

Comment:
Main downside of the separate step is probably that the whole lifecycle (checkout, compile, test, install deploy) gets duplicated. But I would say it's mostly an optimization concern and happy to go with something that works for now.

Reply:
that's the problem. I was debating that point, that it will mean having many ``if:...`` on some several steps